Tasks/ServiceFabricUpdateManifests/Get-VstsBuild.psm1

```diff
@@ -60,30 +64,42 @@
             }
             elseif ($artifact.resource.type -eq "container") # The artifact is in a hosted server and must be downloaded into a temp folder
             {
-                $agentTmpFolder = Join-Path (Get-VstsTaskVariable -Name Agent.WorkFolder -Require) "tmp"
+                $agentTmpFolder = Join-Path $env:Temp $build.buildNumber
```

Comment:
This is problematic. Two builds from different build definitions can have same buildNumber property. BuildId is unique but only within a project so you cannot even use that.